sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -241,12 +236,19 @@ abstract class CodeGenerator[InType <: AnyRef, OutType <: AnyRef] extends Loggin
    */
   protected def compile(code: String): Class[_] = {
     val startTime = System.nanoTime()
+    // Current class loader may be ExecutorClassLoader, which will cause fail to find
+    // class in java.lang.  It's also slow, use the default JVM class loader.
+    val currentThread = Thread.currentThread()
+    val oldClassLoader = currentThread.getContextClassLoader
+    currentThread.setContextClassLoader(getClass.getClassLoader)
```

Comment:
i think we do not need to set setContextClassLoader. and we can set ClassBodyEvaluator's classLoader. code is:
    val classBodyEval =new ClassBodyEvaluator()
      classBodyEval.setParentClassLoader(getClass.getClassLoader)
      classBodyEval.cook(code)
      classBodyEval.getClazz()
some information: http://janino.net/javadoc/org/codehaus/janino/ClassBodyEvaluator.html,
https://github.com/codehaus/janino/blob/6e61d5b71b4e8682a3497cb1e33a005bee4ca0a2/janino/src/org/codehaus/janino/SimpleCompiler.java#L172